core/src/main/java/cz/xtf/core/namespace/NamespaceManager.java

```diff
@@ -0,0 +1,192 @@
+package cz.xtf.core.namespace;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.engine.descriptor.MethodBasedTestDescriptor;
+import org.junit.platform.engine.TestDescriptor;
+
+import cz.xtf.core.config.OpenShiftConfig;
+import cz.xtf.core.context.TestCaseContext;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.core.waiting.SimpleWaiter;
+import io.fabric8.kubernetes.api.builder.Visitor;
+import io.fabric8.kubernetes.api.model.NamespaceBuilder;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class NamespaceManager {
+    /**
+     * By default (xtf.openshift.namespace.per.testcase=false) all entries in map point to value returned by
```

Comment:
Do we need to elaborate implementation details here? Cant we just say it is mapping of namespace to testcase.

Reply:
It's high level describtion of the logic. It actually does not go into detail from my pov :-)

Reply:
Ok. Not insisting on that. 

My only concern was to eliminate mention of `xtf.openshift.namespace.per.testcase` to have it encapsulated on few places. But it is still NamespaceManager where it belongs, so resolving.